common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -831,6 +830,49 @@ protected AccountRecord getCachedAccountRecord(
                         .getAuthorityTypeString()
         );
 
+        AccountRecord targetAccount = getCachedAccountRecordFromCallingAppCache(parameters);
+        if (targetAccount != null) {
+            return targetAccount;
+        } else {
+            Logger.info(TAG, "Account not found in app cache..");
+            targetAccount = getCachedAccountRecordFromAllCaches(parameters);
```

Comment:
Something that probably can be tweaked:

1. rename getCachedAccountRecordFromAllCaches() to getCachedAccountRecordFromCaches()
2. In MsalOAuthTokenCache.getCachedAccountRecordFromCaches(), invoke getCachedAccountRecordFromCallingAppCache() before proceeding with the rest of the logic.
3. In BrokeOAuthTokenCache.getCachedAccountRecordFromCaches(), invoke BrokerUtil.getCacheRecordListFromBrokerCache()

Reasoning: BrokerUtil.getCacheRecordListFromBrokerCache() is already looking into the calling app cache.

Reply:
I think I will keep this as is because BrokeOAuthTokenCache is in common and cannot invoke BrokerUtil.getCacheRecordListFromBrokerCache() which is in broker module. 
Also, I think in getCachedAccountRecordFromAllCaches we are currently doing a lookup with localAccountId specifically whereas in BrokerUtil.getCacheRecordListFromBrokerCache() we are doing a lookup with environment and clientId. So, there is some difference in cache lookup.